lib/pure/pegs.nim

```diff
@@ -86,6 +86,22 @@ type
     else: sons: seq[Peg]
   NonTerminal* = ref NonTerminalObj
 
+proc name*(nt: NonTerminal): string = nt.name
```

Comment:
I think it may be better to just export these fields, why do we need these procs?
Anyway, wait for @Araq or someone else to comment before removing this :)

Reply:
> why do we need these procs?

For read-only access?

Reply:
@data-man well, ok

Reply:
The application I had in mind should just read the Peg nodes, not fiddle around with them. If there are others which need write access, maybe we should indeed just export the fields.

Reply:
I think a foolproof is good.

Reply:
Thinking about it, if we want an "editable" PEG AST, it would probably be better to go all the way (like the Nim AST): make the Node type a ``ref``, define a ``[]`` for the sons and work with getters **and** setters.

Reply:
But if I get a choice, I'll just stick with my getters.

Reply:
I don't see a reason to keep these read-only. But I'll let @Araq decide, this is his module.

Reply:
These getters are much better than full access to the underlying data-structure, see ``packedjson`` about my reasons.